File: backend/internal/models/events.go
package models

// WebSocket Event Constants to satisfy Rule 3 (No Magic Strings)
const (
	EventPartnerJoined = "partner_joined"
	EventPartnerLeft   = "partner_left"
	EventTimerStart    = "timer_start"
	EventTimerSync     = "timer_sync"
	EventTimerComplete = "timer_complete"
	EventPokeSent      = "poke_sent"
	EventPokeReceived  = "poke_received"
)

File: frontend/types/events.ts
// WebSocket Event Constants to satisfy Rule 3 (No Magic Strings)
export const WS_EVENTS = {
  PARTNER_JOINED: "partner_joined",
  PARTNER_LEFT: "partner_left",
  PRESENCE_STATE: "presence_state",
  TIMER_START: "timer_start",
  TIMER_SYNC: "timer_sync",
  TIMER_COMPLETE: "timer_complete",
  SESSION_ENDED: "session_ended",
  POKE_SENT: "poke_sent",
  POKE_RECEIVED: "poke_received",
} as const;

export type WSEventType = typeof WS_EVENTS[keyof typeof WS_EVENTS];
